packages/keyboard-key/src/index.ts

```diff
@@ -0,0 +1,418 @@
+
+    // handle events with a `key` already defined
+    if (isEvent && event.key) {
+      return event.key;
+    }
+
+    // tslint:disable-next-line: deprecation
+    let name = codes[(isEvent ? event.keyCode || event.which : eventOrCode) as number];
+
+    if (Array.isArray(name)) {
+      if (isEvent) {
+        name = name[event.shiftKey ? 1 : 0];
+      } else {
+        name = name[0];
+      }
+    }
+
+    return name;
+  }
+};
+
+const keyboardKey = keyboardKeyDefinition as typeof keyboardKeyDefinition & KeyNames;
+
+// Populate names on keyboardKey.
+for (const code in codes) {
+  if (codes.hasOwnProperty(code)) {
+    const value = codes[code];
+    if (typeof value === 'string') {
+      // tslint:disable-next-line:no-any
+      (keyboardKey as any)[value] = Number(code);
+    } else {
+      // Array of valid values which map to the same code.
+      for (let i = 0; i < value.length; i++) {
+        // tslint:disable-next-line:no-any
+        (keyboardKey as any)[value[i]] = Number(code);
+      }
+    }
+  }
+}
+
+// ----------------------------------------
+// By Alias
+// ----------------------------------------
+// provide dot-notation accessible keys for all key names
+keyboardKey.Spacebar = keyboardKey[' '];
+keyboardKey.Digit0 = keyboardKey['0'];
+keyboardKey.Digit1 = keyboardKey['1'];
+keyboardKey.Digit2 = keyboardKey['2'];
+keyboardKey.Digit3 = keyboardKey['3'];
+keyboardKey.Digit4 = keyboardKey['4'];
+keyboardKey.Digit5 = keyboardKey['5'];
+keyboardKey.Digit6 = keyboardKey['6'];
+keyboardKey.Digit7 = keyboardKey['7'];
+keyboardKey.Digit8 = keyboardKey['8'];
+keyboardKey.Digit9 = keyboardKey['9'];
+keyboardKey.Tilde = keyboardKey['~'];
+keyboardKey.GraveAccent = keyboardKey['`'];
+keyboardKey.ExclamationPoint = keyboardKey['!'];
+keyboardKey.AtSign = keyboardKey['@'];
+keyboardKey.PoundSign = keyboardKey['#'];
+keyboardKey.PercentSign = keyboardKey['%'];
+keyboardKey.Caret = keyboardKey['^'];
+keyboardKey.Ampersand = keyboardKey['&'];
+keyboardKey.PlusSign = keyboardKey['+'];
+keyboardKey.MinusSign = keyboardKey['-'];
+keyboardKey.EqualsSign = keyboardKey['='];
+keyboardKey.DivisionSign = keyboardKey['/'];
+keyboardKey.MultiplicationSign = keyboardKey['*'];
+keyboardKey.Comma = keyboardKey[','];
+keyboardKey.Decimal = keyboardKey['.'];
+keyboardKey.Colon = keyboardKey[':'];
+keyboardKey.Semicolon = keyboardKey[';'];
+keyboardKey.Pipe = keyboardKey['|'];
+keyboardKey.BackSlash = keyboardKey['\\'];
+keyboardKey.QuestionMark = keyboardKey['?'];
+keyboardKey.SingleQuote = keyboardKey["'"];
+keyboardKey.DoubleQuote = keyboardKey['"'];
+keyboardKey.LeftCurlyBrace = keyboardKey['{'];
+keyboardKey.RightCurlyBrace = keyboardKey['}'];
+keyboardKey.LeftParenthesis = keyboardKey['('];
+keyboardKey.RightParenthesis = keyboardKey[')'];
+keyboardKey.LeftAngleBracket = keyboardKey['<'];
+keyboardKey.RightAngleBracket = keyboardKey['>'];
+keyboardKey.LeftSquareBracket = keyboardKey['['];
+keyboardKey.RightSquareBracket = keyboardKey[']'];
```

Comment:
Too late, but this is not treeshakable as it's statics 😭 

But even with `export const RightSquareBracket` will not help much as `getCode()` and `getKey()` functions rely either on `this` or on `codes`.

So `2.4kB` is a price for this module: https://bundlephobia.com/result?p=@fluentui/keyboard-key@0.1.1

Reply:
Is there any way we could break up the functionality into smaller pieces so some of it is tree-shakeable? Or are there any parts of this that could optionally rely on types for compile-time-only verification? (runtime versions would also be available, but could be tree-shaken out if a TS project didn't use them)